docs/source/docs/registry.md

```diff
@@ -1,32 +1,80 @@
-class for use.
-
-With the registry, when developing new classes **inside DeepReg**, we should use the
-corresponding decorator, so that the class is registered. Please check
-`deepreg/model/backbone/u_net.py` as an example. Moreover, the corresponding
-`__init__.py` (maybe multiple ones) should be modified so that these classes will be
-automatically registered when executing `import deepreg`. For defining custom classes
-outside DeepReg, more detailed tutorial will be released later.
-
-For now, we only support custom classes of
-
-- backbone
-- loss
+DeepReg adopts the
+[registry system](https://github.com/DeepRegNet/DeepReg/blob/main/deepreg/registry.py)
+to facilitate the usage of custom functionalities.
+
+## Description
+
+The class `Registry` maintains a dictionary mapping `(category, key)` to `value`, where
+
+- `category` is the class category, e.g. `"backbone_class""` for backbone classes.
+- `key` is the name of the registered class, e.g. `"unet"` for the class `UNet`.
+- `value` is the registered class, e.g. `UNet` corresponding to `"unet"`.
+
+A global variable `REGISTRY = Registry()` is defined to provide a central control of all
+classes.
+
+### Register a class
+
+To register a class into `REGISTRY`, it is recommended to use `register` as a
+**decorator**. Consider the `UNet` class for backbone as an example.
+
+```python
+@REGISTRY.register(category="backbone_class", name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+The decorator will automatically register the class when this class get imported.
+Therefore, in order to make sure this class registered when we `import deepreg`, this
+class and related parent modules needs to be imported in the `__init__.py` files.
+
+For the purpose of code simplification, a specific register function is defined for each
+category. For instance, we can use `register_backbone` for `UNet`:
+
+```python
+@REGISTRY.register_backbone(name="unet")
+class UNet:
+    """UNet-style backbone."""
+```
+
+### Instantiate a class
+
+To instantiate a registered example in `REGISTRY`, we call `build_from_config` , which
+allows creating a class instance from a config directly. This allows for the
+simplification of the config file such that each configuration only needs to provide the
+name of the class and the arguments to build the class for use.
+
+For instance, to instantiate a UNet class,
+
+```python
+unet = REGISTRY.build_from_config(category="backbone_class", config={"name":"unet", **kwargs})
+```
+
+where `kwargs` represents other required arguments.
+
+Similarly, for the purpose of code simplification, a specific build function is defined
+for each category. For instance, we can use `build_backbone` for `UNet`:
+
+```python
+unet = REGISTRY.build_from_config(config={"name":"unet", **kwargs})
+```
+
+## Example usages
+
+To further explain how to use `REGISTRY` for using customized classes, detailed examples
+are provided for each category below.
```

Comment:
where are the examples for each category?

Reply:
as following, "custom backbone", no?

Reply:
https://github.com/DeepRegNet/DeepReg/blob/main/examples/custom_backbone.py

Link will be here, but need to wait until merged.

Reply:
I think maybe consider changing to: 'A detailed example is provided for the `BACKBONE_CLASS` category below' or something similar.

Using 'each category' could be understood as meaning that you have a detailed example for each category i.e. `BACKBONE_CLASS`, `LOSS_CLASS`, `MODEL_CLASS`, `DATA_AUGMENTATION_CLASS` so I think it is best to just specify or change 'each category' to 'some categories' for this sentence.

Reply:
😂oh we should have examples per category, just need to do it.. I see why it's confusing now. Will change.